core/types/calls.cc

```diff
@@ -536,7 +536,8 @@ DispatchResult dispatchCallSymbol(const GlobalState &gs, DispatchArgs args,
                 // catch the special case of `interface!`, `abstract!`, `final!`, or `sealed!` and
                 // suggest adding `extend T::Helpers`.
                 if (args.name == core::Names::declareInterface() || args.name == core::Names::declareAbstract() ||
-                    args.name == core::Names::declareFinal() || args.name == core::Names::declareSealed()) {
+                    args.name == core::Names::declareFinal() || args.name == core::Names::declareSealed() ||
+                    args.name == core::Names::requiresAncestor()) {
```

Comment:
What's the solution when a module must `extend` a class, not `include` it?

Reply:
Ah actually maybe this doesn't make sense because `self.` methods of a module are never mixed into the target module.

nvm!

Reply:
We have a test to check this: https://github.com/sorbet/sorbet/pull/3409/files#diff-862cb4a86571973f989ed33b9bd48785R1-R21

Reply:
yeah i was thinking the other way around, like

```ruby
module A
  def self.foo
    raise
  end
end
```

but this is already fine, with no need for the require_ancestor line.